### DISCUSSION TITLE
What next?

### DISCUSSION THREAD

**Discussion by edwardtfn** [Blackymas/NSPanel_HA_Blueprint#718] [Polls]
Please give your thoughts about where this project should focus next.

I want all of the items and I will try to help in all of those, but if I could choose only one for having first, which one should I focus?

**Comment by thesebastianf** [comment]
different HOME pages to choose from (different tft firmwares)? depending on the focus of the panel.
e.g. - Focus on Weather
      - Focus on Sensor status (hot water etc etc)
      - Focus on alarm control
      - ....

**Reply by edwardtfn** [reply]
Yeah, I've seen people asking for that before and it was a bit hard for me to understand this need, but then I got my own experience pointing to this.
My first panel is right by my front door, so in that one I wanna look the weather before I go outside, then later I got a new panel which is sitting on my desk in the home office so I can control the entire house easily, and, at this one, the weather is not my priority at all. I ended up customising the TFT used in the one by the front door...
So I agree it would be nice to have different profiles... I'm not sure if this is feasible, but would be nice to have.

**Reply by thesebastianf** [reply]
> My first panel is eighth by my front door, so in that one I wanna look the weather before I go outside, then later I got a new panel which is sitting on my desk in the home office so I can control the entire house easily,

which is exactly my use cases as well :-) 
can you share the custom tft? :D

**Reply by edwardtfn** [reply]
I can share, for sure, but before that, I've to give some important advise:
1. This is a customization of this project, therefore, no support will be provided by the project team.
2. This is based on my personal needs, so this might not be exactly what you want and I cannot guarantee all the functionalities of the original TFT provided by this project will be there.
3. I often (including right now) use `dev` environment, so the files might not be compatible with your system.


Currently the files are:
  - nspanel_eu.tft: v3.2.3 (`beta`)
  - nspanel_us.tft: v3.3 (`dev`)

I've replaced the panel from the front door (hall) and I'm not using EU version anymore. This is the reason the EU version is in an "older" format.

This is my working repository where you will find the files I'm currently using: https://github.com/edwardtfn/HomeAssistant-Config/tree/main/www/nspanel/hall

**Reply by edwardtfn** [reply]
I've just merged #898 to `dev`. Hopefully that will come in a next release soon.
With that you can select other pages to be shown when the panel wakes up, which might be a step towards what you are asking for.

**Comment by JUK-Bavaria** [comment]
A small wish from me:
I use the entity screen to check which Windows/Door is open in detail. Because this are binary sensors, it will be displayed as on/off. It would be nice, if there is possibility to change this, i.e. open/closed (or for me in German Offen/Zu). That would be good for the WAF. ;)

**Reply by edwardtfn** [reply]
Not impossible, probably not that hard to implement, but that would require translations to all the different languages and ideally to all the different [device classes supported by binary sensors](https://www.home-assistant.io/integrations/binary_sensor/#device-class).

**Reply by JUK-Bavaria** [reply]
Maybe there is the possibility to use variables, so the user is able to overwrite on/off with his own words?

**Reply by adamengineering** [reply]
I've had the same requirement and solved it using sensors in HomeAssistant, that return the value I want for oo/off using value_template.

`garage_door_sensor: value_template: >- {% if is_state("binary_sensor.garagedor","on")%} open {% else %} closed {% endif %}`

Instead of using the original binary_sensor I added the 'template' one to the NSPanel

**Comment by rleongcs** [comment]
Not sure if this is possible on the panel...CCTV feed?

**Reply by deejaybeam** [reply]
> Not sure if this is possible on the panel...CCTV feed?

no - live-stream in any way is not possible

**Reply by rleongcs** [reply]
How about a snapshot then that can be sent to the panel when motion detected?

**Reply by edwardtfn** [reply]
How would you use this? I wanna hear more about your use case.

**Reply by deku-m** [reply]
> How would you use this? I wanna hear more about your use case.

Frigate for example provides this with detecting a person/object. It takes a screenshot of the moment the person comes into vision.
These are created as only a snapshot as image format and shown.  The same can then be used for when the doorbell rings a screenshot will be taken and provided on a page on the NSpanel.

See example:
![afbeelding](https://user-images.githubusercontent.com/37069737/234533609-561a36da-dcbd-4e6d-95e8-674b36fcafe0.png)

**Reply by rleongcs** [reply]
Yes...my use case is as @deku-m outlined. I am not using frigate for the moment but just the camera detected motion and capture a snapshot and I would like that it can be displayed to the NSPanel as an alert. Very good use of the NSPanel especially the one beside the front door.

**Reply by deejaybeam** [reply]
> How about a snapshot then that can be sent to the panel when motion detected?

This is also not possible

When compiling the tft (from the HMI) all grafics are included in the nextion firmware an it´s not possible to send a pic, or a stream to that kind of displays

**Reply by deku-m** [reply]
> > How about a snapshot then that can be sent to the panel when motion detected?
> 
> This is also not possible
> 
> When compiling the tft (from the HMI) all grafics are included in the nextion firmware an it´s not possible to send a pic, or a stream to that kind of displays

It cant ask for the snapshot stored on web www folder on HA and then be displayed by using a entity like Home Assistant does?

**Reply by deejaybeam** [reply]
> > > How about a snapshot then that can be sent to the panel when motion detected?
> > 
> > 
> > This is also not possible
> > When compiling the tft (from the HMI) all grafics are included in the nextion firmware an it´s not possible to send a pic, or a stream to that kind of displays
> 
> It cant ask for the snapshot stored on web www folder on HA and then be displayed by using a entity like Home Assistant does?

no

**Reply by rleongcs** [reply]
That's a real bummer. Looks like I would need the NSPanel Pro/Tuya Andriod Panel in order to do that.

**Reply by deejaybeam** [reply]
> That's a real bummer. Looks like I would need the NSPanel Pro/Tuya Andriod Panel in order to do that.

yes
I know that´s bad, and there are nextion displays with this functionality, but not in NSPanel

**Comment by zwevert** [comment]
A way to (child)lock the device would be awesome.

**Comment by Hypernova007** [comment]
It would be great if the cover page would support the Tilt function.  The possibility to add extra buttons on the cover page would be nice to, this can be used for preconfigured positions (for instance: 70% open)

**Comment by thesebastianf** [comment]
![image](https://user-images.githubusercontent.com/13987575/235462752-16b05dec-bcb8-4f9c-a560-54825969cd6d.png)

**Reply by edwardtfn** [reply]
About "Color depending on status of an entity", isn't that way already?
That bar will be gray or blue depending on the status of the entity associated to the hardware buttons, while the relays 1 and 2 are connected to the numbers 1 and 2 in the top left corner of the screen.
In the buttons pages also the icon colors will change based on the status of the entity.

**Reply by thesebastianf** [reply]
> About "Color depending on status of an entity", isn't that way already?
> That bar will be gray or blue depending on the status of the entity associated to the hardware buttons, while the relays 1 and 2 are connected to the numbers 1 and 2 in the top left corner of the screen.
> In the buttons pages also the icon colors will change based on the status of the entity.

In my Case I Open the Garage door.
So entity Switch is Just triggering the actor.
As Status if the Garage is actually Open I would need another enetities state of a Reed contact.

**Reply by edwardtfn** [reply]
Have you thought about setting your garage door as a template cover in order to link that to the panel? The advantage is to have the states `opening` and `closing` in addition to `open` and `closed` available in a door or `on`/`off` in a switch.
We currently don't have different colors for `opening` and `closing` (I'm pretty sure it will show like `open` and `closed` respectively), but certainly would be much easier to implement a new behavior. And you probably will have a better experience in your dashboards.

Edit: I just saw there is even an example using for a garage dorr in the template cover docs: https://www.home-assistant.io/integrations/cover.template/#garage-door

**Comment by Doughboy68** [comment]
Please add fan speed control. Even if only be able to add a percent slider (like a light dimmer) and not the OFF, LO, MED, HIGH options that would still be OK. I have a bedroom fan that has these speeds, but also could just use slider to set speed as option... better than just on/off.

**Reply by edwardtfn** [reply]
Similar requests:
- #404 
- #791

**Reply by edwardtfn** [reply]
I've just submitted #897 which will add a fan speed control page. Hopefully this gets all the way to the next release to `main`.

**Comment by en2te** [comment]
Entity values / templates in notifications would be nice. That would be a great feature to display scraped news or sensor-values in notifications.

**Reply by edwardtfn** [reply]
Can't you do do that today? I'm pretty sure something like this would work:

```yaml
service: esphome.office_workstation_panel_notification_show
data:
  label: '{{ states("sensor.vasttrafik_departure_table") }}'
  text: '{{ states_attr("sensor.vasttrafik_departure_table", "departure") }}'
```

**Reply by en2te** [reply]
Yes, works. Thanks. Maybe i had a problem with the sensor-data when i tried.

**Comment by rudgej** [comment]
I know most of these panels have weather / time on the Home Screen but they are really just fillers. The panel is used to control automations / lights and all the Home Screen does is put an additional swipe in the way of getting that job done. Could there be an option that says what of the current screens ( home / button 1/2/3/4) you want as the default ? 

That way it could default to the switches / automations for that room. I have 10 panels running

**Reply by edwardtfn** [reply]
Wow! 10 panels!? This is impressive!

This request looks similar to this: https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/718#discussioncomment-5715670

**Comment by Rainerlan** [comment]
Entity pages should update the entity status while the page is being displayed - and the entity pages may use/display "entity-page"-icons from lovelace panel where a state-dependent icon is being displayed instead of the entity status (text).

On the entity pages which I have defined, where all door/window-sensor status (open/closed) are being listed, 
I found that the status of the entities is not being updated, while the page is being displayed.
Also the status is just displayed as "text" - opened/closed (Not by an icon - eg. a slider).

Also a configurable timeout per entity page would be nice, too.

**Reply by Blackymas** [reply]
Updating the entity status does not make sense for us for 2 reasons

1. it causes extreme load for NSpanel, HA and the network

2. the normal user stands about 20-30 seconds in front of it to check the status. No one stands in front of it changes something and looks if it changes on the NSPanel.


Therefore, we have decided against it and it is currently not planned :)

**Reply by edwardtfn** [reply]
Auto-update entities page was handled by #819 which is part of v3.4 to be released very soon.
The load concerns have been taken in account and I don't think that will be an issue with this implementation as those elements are updated only when one of the entities pages are visible. 😉

**Reply by edwardtfn** [reply]
How and what is displayed, however, isn't changing. It's hard to add any control as there are limitation on the display, where we need pre-defined fields do be hard coded and there are also some difficulties to show anything different than what it is stored in HA back-end as that would require a huge front end management, translations, etc.

**Reply by edwardtfn** [reply]
> Also a configurable timeout per entity page would be nice, too.

This requires some re-design, but it is feasible. Could you please provide more context? Why is this needed?

**Comment by SmartHomeAu** [comment]
Thank you so much to the team for this awesome creation! Media player controls would be simply awesome.  I know this is a feature being worked on :). Is there a rough timeline for this?

Donation coming your way.  Thanks again!

**Reply by Blackymas** [reply]
Hey, no we have no Timeline :)

**Comment by edwardtfn** [comment]
v4.0 is on `beta` now and will be released soon. With that, most of the items on the original topics are delivered (at least in a minimum level) and this topic is obsolete now.
I will close this discussion and as you to add new ideas as [new "Feature request"](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/new).
